lib/mongoid/history/trackable.rb

```diff
@@ -71,6 +71,16 @@ def history_tracks
           @history_tracks ||= Mongoid::History.tracker_class.where(scope: related_scope, association_chain: association_hash)
         end
 
+        def undo(modifier = nil, options_or_version = nil)
+          versions = get_versions_criteria(options_or_version).to_a
+          versions.sort! { |v1, v2| v2.version <=> v1.version }
+
+          versions.each do |v|
+            undo_attr       = v.undo_attr(modifier)
+            self.attributes = undo_attr
+          end
+        end
+
```

Comment:
It's extremely similar to `undo!`. I think you need to rewrite `undo!` as `undo` and `save!`.